### ISSUE TITLE
Temperature stability

### ISSUE THREAD

**Issue by avrosmalen** [kaytat/stoker#2]
Hi there!

I can't find another way to contact you so hence this message.

I am new to this whole GutHub thing but I would like to help out on the Stoker firmware. Right now the temperature varies quite a bit. It looks like the blower is engaged when a certain temperature threshold is passed and shut down when an upper thershold is passed. When the blower is disengaged the temperature continues to rise for a while. After this the smokers gradually starts to cool but it takes a while for temperature to climb again after the blower is engaged. In other words the temp will vary which is is sort of contrary to the intended use. I think a better algorithm for controlling temperature can easily fix this.

The issue is discussed here: http://amazingribs.com/bbq_equipment_reviews_ratings/thermometers/rocks-bar-b-que-stoker

I would like to help out but it has been a while since I was programming. Solving this issue however has more to do with logic than the latest and greatest on programming but more on physics and math.

Could you share the code with me, especially the temperature controlling bit so I can think along and improve the Stoker?

Thanks

Anton

**Comment by kaytat** [comment]
Hi Anton,

I've been trying hard to share the code, but some of it is protected by 
other parties and they have been unresponsive.

I can see if I can only release the control part.

FYI - it's a PID algorithm.

Thanks,
Ken

**Comment by avrosmalen** [comment]
Thanks Ken,

I will look in to it!

Update: I cannot find code, am I looking in the right place or have you not been able to release this part of the code yet?

I was expecting a PID algorithm, because of the sort of response it looks like only the proportional gain is used. Is this correct?

Cheers

Anton

**Comment by BrunDog** [comment]
Hi, Hope you don't mind the contact... with Stoker OOB, is there any chance the Stoker code could be released, so we don't have dead units?

**Comment by kaytat** [comment]
Yeah, looking into it. It's based on the TINI OS source code from Maxim/Dallas Semi and I'm still exploring options regarding publishing my changes.

**Comment by RJ-Make** [comment]
Any progress on this @kaytat ?

Thanks for all your help

**Comment by BrunDog** [comment]
Likewise, I'd be interested in having access so we can keep our hardware working in the future.

**Comment by BrunDog** [comment]
Hi Ken,

Touching base on this again... in its current iteration, I can't connect my unit to WiFi due to security evolution of WiFi protocols.

If you can possibly provide the code, that would be great. If not, can you at least share what you know about the hardware (micro-controller, temp probe communications, etc.)?
